plugins/pay.c

```diff
@@ -1093,6 +1094,90 @@ static struct command_result *handle_paystatus(struct command *cmd,
 	return command_success(cmd, ret);
 }
 
+static const jsmntok_t *copy_member(char **ret,
+				    const char *buf,
+				    const jsmntok_t *result,
+				    const char *membername,
+				    const char *term)
+{
+	const jsmntok_t *m = json_get_member(buf, result, membername);
+	if (m)
+		tal_append_fmt(ret, "'%s': %.*s%s",
+			       membername,
+			       json_tok_full_len(m), json_tok_full(buf, m),
+			       term);
+	return m;
+}
+
+static struct command_result *listpayments_done(struct command *cmd,
+						const char *buf,
+						const jsmntok_t *result,
+						char *b11str)
+{
+	size_t i;
+	const jsmntok_t *t, *arr;
+	char *ret;
+	bool some = false;
+
+	arr = json_get_member(buf, result, "payments");
+	if (!arr || arr->type != JSMN_ARRAY)
+		return command_fail(cmd, LIGHTNINGD,
+				    "Unexpected non-array result from listpayments");
+
+	ret = tal_fmt(cmd, "{ 'pays': [");
+	json_for_each_arr(i, t, arr) {
```

Comment:
Am I correct in thinking that this loop will eventually coalesce all parts of a multi-part payment?

Reply:
Yes.